gammapy/estimators/map_profile.py

```diff
@@ -0,0 +1,313 @@
+"""Tools to create profiles (i.e. 1D "slices" from 2D images)."""
+import numpy as np
+from astropy import units as u
+from astropy.wcs.utils import proj_plane_pixel_scales
+from astropy.coordinates import Angle, SkyCoord
+from regions import RectangleSkyRegion, PixCoord
+from gammapy.maps import MapAxis
+from gammapy.utils.table import table_from_row_data
+from gammapy.estimators import ImageProfile
+from gammapy.stats import WStatCountsStatistic, CashCountsStatistic
+from gammapy.datasets import SpectrumDatasetOnOff
+
+__all__ = ["MapProfileEstimator", "make_orthogonal_boxes"]
+
+
+class MapProfileEstimator:
```

Comment:
To keep consistency within our estimator naming I would actually call this thing `ExcessProfileEstimator`, (or `ExcessMapProfileEstimator`, if the result is a `MapProfile`) because it's equivalent in (algorithmic) functionality to the `ExcessMapEstimator` we already have.

Reply:
Well, naming is always subjective...
This estimator does far more than making a profile of excess: counts, background, excess, flux, brightness.
The name used here includes Map as it takes a MapDataset object in input....
And the name of the file is coherent with the name of the class.

However, one can change the name of the class, but to be coherent the name of the file also (leading to the close of this PR maybe, and the opening of an other one...).

Any advice?

Reply:
@bkhelifi I mostly agree that the choice of naming schemes is subjective, however once a scheme is chosen I think it should be consistent within the software. Otherwise maintenance as well as usability is affected. In this case the functionality is very close to the `ExcessMapEstimator`, which also does much more (like computing counts, background, etc. ). 

That's why I said in one of my other comments, whatever we do, it should just be consistent between the `ExcessMapEstimator` and `ExcessProfileEstimator`. That's all I care about. If you have a better suggestion for a name I'd be open. But you have to convince @registerrier on the renaming of the `ExcessMapEstimator` then...:-)

Reply:
Done